CMakeLists.txt

```diff
@@ -434,7 +435,20 @@ endforeach(ext)
 
 add_subdirectory (frontends)
 add_subdirectory (midend)
-add_subdirectory (control-plane)
+
+if(NOT BUILD_CONTROL_PLANE)
```

Comment:
This is a good first step, but I would move these checks into CMakelists of the respective back end folder. 

In all likelihood things will break if you try to build without control-plane now, because there are many CMake targets that depend on it. There might even be some code that accidentally includes it. 

To fix this for CMAKE you can at least create a variable (e.g., P4C_CONTROLPLANE) which is only set with the value `controlplane` when the control plane is active.

Reply:
But that won't solve our problem at the moment. I looked all over Backends, if am not wrong it's p4tools, dpdk and Tests which requires control-plane as dependency.

Reply:
What do you mean? These error checks should be handled in each individual back end.

Reply:
Ohhh I get it I should have handled this in the individual backend. Also at the moment 3 backends are there having Control-plane as dependency if I am not wrong?

Reply:
There might be more, but we will see this once we try to start to compile code without the control plane enabled.

Reply:
Okay I'll try to move these checks to the individual backend ✅